bitcoin/src/consensus/serde.rs

```diff
@@ -104,10 +104,32 @@ pub mod hex {
     #[derive(Debug)]
     pub struct DecodeInitError(hashes::hex::Error);
 
+    impl fmt::Display for DecodeInitError {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            write!(f, "creation of hex string decoder failed")
+        }
+    }
+
+    #[cfg(feature = "std")]
+    impl std::error::Error for DecodeInitError {
+        fn source(&self) -> Option<&(dyn std::error::Error + 'static)> { Some(&self.0) }
+    }
+
     /// Error returned when a hex string contains invalid characters.
     #[derive(Debug)]
     pub struct DecodeError(hashes::hex::Error);
 
+    impl fmt::Display for DecodeError {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            write!(f, "hex string contains invalid characters")
```

Comment:
`write_err`

Reply:
I'm not totally sure on the best content to put in the error message of `write_err`, noting it here but can discuss on #1862.